index.js

```diff
@@ -44,8 +37,15 @@ function mergeLocation (source, dest) {
   }
 }
 
+let isLong
+
 function build (schema, options) {
   options = options || {}
+
+  if (options.useLong) {
```

Comment:
This should be enabled by default unfortunately :(, otherwise it'd be semver-major.

Reply:
I suggest making it a major change.

Reply:
I'm ok with it.. unless you plan to use this in Fastify v3.

Reply:
major update seems to be the one possible imo, even if  we enable it by default it will be breaking changes for users without `long` package installed

Reply:
I mean that this should be enabled by default if long is available, with an option to disable it entirely.

Reply:
We can add:

```js
let useLong = options.useLong === undefined ? Boolean(require.resolve('long')) : options.useLong;

if (options.useLong) {
  isLong = require('long').isLong
}

```

if the goal is to make it backwards compatible.

Reply:
Never mind. I didn't realize that `require.resolve` throws an error if it cannot resolve. That gets us to where we started.

Reply:
`require.resolve` will throw as well, see https://nodejs.org/api/modules.html#modules_require_resolve_request_options

Reply:
Adding a try/catch inside build seems ok to me. Would that fix your problem?

Reply:
The annoying part are the warnings that appear in the build log when using the `try..catch` error.

Overall, seems low-value reason to keep compatibility with. The fact that library changes behavior simply based on whether a dependency exists (somewhere in the reachable dependency tree) seems very wonky.

I advocate for major version bump.

Reply:
You're saying this creates an error log?:

```js
try {
  useLong = Boolean(require.resolve('long'))
} catch {
  useLong = false
}
```